### DISCUSSION TITLE
How do I group all updates per manager for all projects using that manager?

### DISCUSSION THREAD

**Discussion by janpieterz** [renovatebot/renovate#25351] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovatebot/github-action@v39.0.1

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have a repository with both C# (Nuget) and Typescript (NPM) projects, distributed through a folder structure. Multiple `csproj` files and multiple `package.json` files define the dependencies.

I'd like to group all Nuget dependencies in one pull request and all NPM dependencies in one pull request, but no permutation of the grouping seems to trigger this.

### What you tried so far.

Tried a variety of package rules:

```
    "packageRules": [
        {
            "groupName": "all dependencies",
            "groupSlug": "all",
            "matchPackagePatterns": ["*"]
        }
    ],
```

```
 "packageRules": [
        {
            "matchManagers": [
                "docker-compose",
                "npm",
                "nvm",
                "github-actions",
                "dockerfile",
                "regex",
                "nuget"
            ],
            "groupName": "{{manager}}",
            "excludePackageNames": ["Microsoft.Azure.Cosmos"]
        }
    ],

```

```
 "packageRules": [
        {
            "matchPackageNames": ["*"],
            "matchUpdateTypes": ["major", "minor", "patch", "pin", "digest"],
            "groupName": "{{dep.manager}}",
            "groupNameTemplate": "${dep.manager}",
            "enabled": true
        }
    ],
```

**Comment by rarkins** [comment]
Simply `"groupName": "{{manager}}"` should work, please focus on that solution alone and provide either full debug logs or a reproduction repo if you can't get it to work

**Reply by janpieterz** [reply]
Thanks! That does seem to group by manager, but doesn't group all the dependencies, so we get a pull request per dependency instead of simply 1 nuget pull request for all nuget dependencies over all projects. 
And it creates two pull requests for a dependency as well, oddly, looks like it was grouped by project name (two projects called `IntegrationTests` and two projects called `UnitTests`. 

Looks like there's some confidential info in the logs, I'll work on a reproduction repo.

**Reply by janpieterz** [reply]
Alright, so I have a reproduction [here](https://github.com/janpieterz/renovate-manager-sample).

I use the same Github Action version for Renovate, and same config as our internal repo but had to add an exclude or match configuration as for some reason it kept exiting early with this error (same config, same workflow file does **not** trigger this for the other repo, very odd.

```
"errors": [
         {
           "topic": "Configuration Error",
           "message": "packageRules[0]: Each packageRule must contain at least one match* or exclude* selector. Rule: {\"groupName\":\"{{manager}}\"}"
         }
       ]
```

Output you can see [here](https://github.com/janpieterz/renovate-manager-sample/pulls), the behavior is different than our repo because of dependencies but I'm looking to these two prs, one for nuget and one for npm, each with all dependencies for all projects listed in each pull request, just separated by manager.

**Reply by janpieterz** [reply]
Debug logs

[renovate_logs.txt](https://github.com/renovatebot/renovate/files/13079597/renovate_logs.txt)

**Reply by rarkins** [reply]
Please reduce the reproduction config to minimal. There are fields like `username` which aren't even valid

**Reply by janpieterz** [reply]
Reduced to be quite minimal, results in [this pull request](https://github.com/janpieterz/renovate-manager-sample/pull/10) with an error about npm, it looks like it's including npm issues and updates into that pull request. Bit tough to digest the logs, but if I manually update react scripts and typescript it works fine 🤔 

[renovate_logs.txt](https://github.com/renovatebot/renovate/files/13125391/renovate_logs.txt)

**Reply by janpieterz** [reply]
On our private repo I get a successful pull request (no NPM errors) but the pull request groups everything together in one pull request, similar as it seems to do in above logs. The pull request is named `update npm` but includes nuget, GitHub actions etc.

**Reply by rarkins** [reply, marked as answer]
This config seems to do what you want: https://github.com/renovate-reproductions/25351/blob/main/.github/renovate.json

FWIW I never recommend setting separateMajorMinor=false

**Reply by janpieterz** [reply]
Looks like it works! Thanks @rarkins .
